### ISSUE TITLE
MODS config: check for presence of existing parent element

### ISSUE THREAD

**Issue by bondjimbond** [mjordan/islandora_oai_download_link#3]
The default MODS element is `<location><url access="download">`, which is perfectly acceptable. However, if there's already a `<location>` element in the MODS record, the new element ends up in a separate <location> parent, which is [bad MODS](https://www.loc.gov/standards/mods/userguide/location.html).

There should ideally be a check for (1) if the user has configured their download link element to be a child of another element (as with the default), then (2) whether the parent element (e.g. <location>) already exists. If (2), the link element should be a child of the existing parent. If not, the full configured element set should be used.

The above may not have been written very well, but I've tried to make it clear.

**Comment by mjordan** [comment]
Copy and pasting a comment from the original [JIRA ticket](https://jira.duraspace.org/browse/ISLANDORA-2007) that spawned this module. An added complication in this situation is that since users can configure their own template, we'd have to make the consolidation code accommodate any element the user configures.

**Comment by mjordan** [comment]
@bondjimbond are you sure repeating `<location>` is bad MODS? https://www.loc.gov/standards/mods/userguide/dlfaquifer.html says that `<location>` is repeatable.

**Comment by bondjimbond** [comment]
I gather from the guidelines that you'd only repeat <location> if you wanted to repeat certain subelements; otherwise you'd group all subelements into one <location>. Could be wrong of course. Going off the examples, as DLF is not super explicit.

**Comment by mjordan** [comment]
I agree we want to share the best possible MODS we can, but I want to be cautious about investing a lot of effort in adhering to sometimes unclear best practice guidelines until we have a real use case to motivate and focus us. I'd rather put effort into tackling #4 since it is a more general problem but also since it might, with a suitable XSLT we could provide, address this issue.

Do we know if any of the harvesters of MODS care if there is an extra, tagged on element with the direct download URL in it? Do they recommend or require adherence to DLF's shareable metadata guidelines? @bradspry, [you said in the JIRA ticket](https://jira.duraspace.org/browse/ISLANDORA-2007?focusedCommentId=56259&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-56259) that DPLA harvests your MODS - what do they require?  (Sorry, a quick web search didn't yield instant gratification on their requirements.)

**Comment by mjordan** [comment]
@bradspry can you point us to any docs on what DPLA recommends in MODS it harvests WRT the download URL for an object?
